packages/core/src/routes/log.ts

```diff
@@ -35,4 +39,36 @@ export default function logRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.get(
+    '/dashboard',
+    koaGuard({
+      query: object({
+        start: string().regex(dateRegex),
+        end: string().regex(dateRegex),
+      }),
+    }),
+    async (ctx, next) => {
+      const {
+        query: { start, end },
+      } = ctx.guard;
+
+      const startTime = new Date(start).valueOf();
+      const endTime = new Date(end).valueOf();
+      assertThat(startTime <= endTime, 'dashboard.wrong_date_range');
+
+      // Convert date closed interval to time left-closed right-open interval:
+      // e.g. [2022-05-01, 2022-05-02] -> [2022-05-01 00:00:00.000, 2022-05-03 00:00:00.000)
+      const endTimeExclusive = endTime + millisecondsInOneDay;
+      const dnuCounts = await getDnuCountsByTimeInterval(startTime, endTimeExclusive);
+      ctx.body = {
+        // DNU: Daily New User
+        dnuCounts,
+        // DAU: Daily Active User
+        dauCounts: {},
```

Comment:
TODO?

Reply:
yes. added issue. (since gao said if if added the issued, will not need to add todo comment.)